src/server/gc_worker/gc_manager.rs

```diff
@@ -449,12 +453,33 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine> GcMan
         // The following loop iterates all regions whose leader is on this TiKV and does
         // GC on them. At the same time, check whether safe_point is updated
         // periodically. If it's updated, rewinding will happen.
+
+        let (tx, rx) = mpsc::channel();
+        let cb = Box::new(move |res| {
```

Comment:
how about using https://docs.rs/crossbeam/latest/crossbeam/sync/struct.WaitGroup.html or `CallbackWaitGroup` of backup-stream util

Reply:
This won't work well because it would require to wait for all tasks finished. 
In the end I used condvar to implement this functionality.